tests/scenario/test_extension_providers.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+from scenario import State
+
+from charm import IstioCoreCharm, flatten_config
+
+#############################################
+# Tests for flatten_config
+#############################################
+
+
+def test_flatten_config_with_flat_dict():
```

Comment:
nit: I'd put this top group in tests/unit/test_extension_providers.py since they aren't scenario tests